src/Reflection/ReflectionClass.php

```diff
@@ -1238,4 +1244,17 @@ public function removeProperty($propertyName)
 
         return false;
     }
+
+    private function findTypeFromAst($namespace, $locatedSource, $type)
```

Comment:
Currently this method is repeated four times, (`ReflectionClass`, `AbstractReflectionFunction`, `ReflectionParameter` and `ReflectionMethod` ? ). We can either create a new base class, or turn the `ContextFactory` into a `TypeFactory`...